actors/market/src/lib.rs

```diff
@@ -811,7 +781,7 @@ impl Actor {
                     }
 
                     let (slash_amount, remove_deal) =
-                        st.process_deal_update(rt.store(), &state, &deal, curr_epoch)?;
+                        st.process_deal_update(rt.store(), &state, &deal_proposal, curr_epoch)?;
 
                     if slash_amount.is_negative() {
```

Comment:
Move this into process_deal_update, if it makes sense at all. Remove it from here and PDU.